cmd/agent/service_windows.go

```diff
@@ -0,0 +1,82 @@
+// +build windows
+
+package main
+
+import (
+	"flag"
+	util_log "github.com/cortexproject/cortex/pkg/util/log"
+	"github.com/grafana/agent/pkg/config"
+	"log"
+	"os"
+	"time"
+
+	"golang.org/x/sys/windows/svc"
+)
+
+const ServiceName = "Grafana Cloud Agent"
+const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+
+type AgentService struct {
+}
+
+func (m *AgentService) Execute(args []string, serviceRequests <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	changes <- svc.Status{State: svc.StartPending}
+
+	// Executable name and any command line parameters will be placed into os.args, this comes from
+	// registry key `Computer\HKEY_LOCAL_MACHINE\SYSTEM\ControlSet001\Services\<servicename>\ImagePath`
+	// oddly enough args is blank
+	fs := flag.NewFlagSet(args[0], flag.ExitOnError)
+	cfg, err := config.Load(fs, os.Args[1:])
```

Comment:
Something seems wrong here mixing both `args[0]` and `os.Args[1:]`. Did you mean to just use one?

Looking at the docs for `golang.org/x/sys/windows/svc
`, it seems like `os.Args` doesn't get modified anywhere by `golang.org/x/sys/windows/svc`

Reply:
Good catch os.Args is the correct one, changing now